packages/docmanager/src/dialogs.ts

```diff
@@ -295,6 +296,12 @@ namespace Private {
     });
 
     label.textContent = trans.__("Don't ask me again");
+
+    label.addEventListener('click', function () {
+      checkbox.checked = !checkbox.checked;
+      manager.nameFileOnSave = !checkbox.checked;
```

Comment:
Wondering whether it would be possible to keep the same functionality and UX, but without having to store state on the doc manager?

Setting `manager.nameFileOnSave` directly might be misleading for example when the checkbox is ticked, but the dialog is "canceled" by clicking outside of it:

https://user-images.githubusercontent.com/591645/122191809-57979b80-ce93-11eb-86c5-3b461486af70.mp4

In that case one could expect the change to not have taken effect since "Enter" was not clicked.

Reply:
Maybe it's possible to keep the state in the rename dialog instead, and wait for Enter to be clicked before deciding whether or not to apply the settings (by checking `result.value`)?

And maybe this logic could then be moved to the plugin level in the `docmanager-extension` package, removing the need for having `nameFileOnSave` on the doc manager.